query/executor.go

```diff
@@ -380,7 +380,7 @@ LOOP:
 	}
 
 	// Send error results for any statements which were not executed.
-	for ; i < len(query.Statements)-1; i++ {
+	for i++; i < len(query.Statements); i++ {
```

Comment:
I ran your new tests and they pass without this change, which is concerning

Reply:
Have simplified tests conditions, complicated test failure modes. Now can differentiate between previous and new code, as well as correcting a few other small bugs.